CMDCT-3764: Add New Parity Section to MCPAR

karla-vm opened the pull request:
### Description
<!-- Detailed description of changes and related context -->
Adding a new topic `Topic XII. Mental Health and Substance Use Disorder Parity` under `Section C: Program-Level Indicators`. Additionally, a new "Parity" radio button option for `VIII: Sanctions`under `Section D. Plan-Level Indicators` has been added.

### Related ticket(s)
<!-- Link to related ticket(s) or issue(s) -->
<!-- Hint: Type CMDCT-<ticket-number> for autolinking -->
CMDCT-3764

---
### How to test
<!-- Step-by-step instructions on how to test, if necessary -->
**Note:** These changes are feature flagged for the July MCPAR release, so you'll need to go to [this line](https://github.com/Enterprise-CMCS/macpro-mdct-mcr/blob/292ce46501725e331836cab8702b5adb60070c96/services/ui-src/src/components/modals/AddEditReportModal.tsx#L44) and set it to `true` to see the updated content.

- Log in as a state user
- Create a new MCPAR report

If you haven't added a plan, you should see the following error message under `/mcpar/plan-level-indicators/sanctions`:

![Screenshot 2024-07-02 at 2 23 53 PM](https://github.com/Enterprise-CMCS/macpro-mdct-mcr/assets/99458559/0f5abdc3-0b04-4a54-a7d7-b12f79f58504)

- Navigate to `/mcpar/program-level-indicators/mental-health-and-substance-use-disorder-parity` and you should see the following page:

<img width="828" alt="Screenshot 2024-07-01 at 2 33 28 PM" src="https://github.com/Enterprise-CMCS/macpro-mdct-mcr/assets/99458559/319184ae-fdc3-475f-9a7c-cc52339c6c84">

- Click on "Yes" to see the nested Parity form and make sure it matches Figma
- Navigate to `/mcpar/plan-level-indicators/sanctions` and you should see the following options for adding a new sanction:

![Screenshot 2024-07-01 at 2 33 57 PM](https://github.com/Enterprise-CMCS/macpro-mdct-mcr/assets/99458559/736b19c3-3adf-4e58-bf1c-238f2dbbcda4)

### Notes
<!-- Changed dependencies, .env files, configs, etc. -->
<!-- Instructions for local dev, e.g. requires new installs in directories -->
`mcpar.json` is updated

---
### Pre-review checklist
<!-- Complete the following steps before opening for review -->
- [ ] I have added [thorough](https://shorturl.at/aejkF) tests, if necessary
- [ ] I have updated relevant documentation, if necessary
- [ ] I have performed a self-review of my code
- [ ] I have manually tested this PR in the deployed cloud environment

---
### Pre-merge checklist
<!-- Complete the following steps before merging -->

#### Review
- [ ] Design: This work has been reviewed and approved by design, if necessary
- [ ] Product: This work has been reviewed and approved by product owner, if necessary

#### Security
_If either of the following are true, notify the team's ISSO (Information System Security Officer)._

- [ ] These changes are significant enough to require an update to the SIA.
- [ ] These changes are significant enough to require a penetration test.
---

<!-- If deploying to val or prod, click 'Preview' and select template -->
_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_

britt-mo reviewed (commented):
Tested locally and I wondered if we need a card to hide the `add sanction` button and display a similar message on the `Sanctions` page to that of `Program Integrity` when no plans have been created?

karla-vm commented:
> Tested locally and I wondered if we need a card to hide the `add sanction` button and display a similar message on the `Sanctions` page to that of `Program Integrity` when no plans have been created?

that's a great question, I brought it up to Design on Slack! thanks for pointing that out, Britt.

britt-mo reviewed (dismissed):
Tested, looks good.

bangbay-bluetiger commented on services/ui-src/src/components/reports/ModalDrawerReportPage.tsx:
```diff
@@ -46,6 +48,14 @@ export const ModalDrawerReportPage = ({ route, validateOnRender }: Props) => {
 
+  // check if there is at least one (1) plan prior to being able to enter Sanctions
+  const checkForPlans = () => {
+    if (entityType === entityTypes[4]) {
```
Would be better if `sanctions` itself were a constant.

bangbay-bluetiger commented on services/ui-src/src/components/reports/ModalDrawerReportPage.tsx:
```diff
@@ -170,36 +180,45 @@ export const ModalDrawerReportPage = ({ route, validateOnRender }: Props) => {
+            >
+              {verbiage.addEntityButtonText}
+            </Button>
+            {reportFieldDataEntities.length !== 0 && (
```
`> 0` a little easier to read than `!== 0`.

bangbay-bluetiger commented on services/app-api/utils/formTemplates/formTemplateModifiers.ts:
```diff
@@ -0,0 +1,69 @@
+        removeAppealsAndGrievances(route.children);
+
+      // remove new radio button option from question D3.VIII.2 in template
+      route.children[5].modalForm.fields[1].props.choices.splice(6, 1);
```
Would selecting the choice to remove by its content instead of index be a little safer?

bangbay-bluetiger commented on services/app-api/utils/formTemplates/formTemplateModifiers.ts:
```diff
@@ -0,0 +1,69 @@
+      );
+
+      // remove new radio button option from question D1.X.6 in template
+      route.children[6].drawerForm.fields[5].props.choices.splice(1, 1);
```
Would selecting the choice to remove by its content instead of index be a little safer? In case something gets moved?

britt-mo reviewed (approved):
Tested, thank you for updating with fixes to the MACPAR that aren't a part of the parity work!

bangbay-bluetiger commented on services/app-api/utils/formTemplates/formTemplateModifiers.ts:
```diff
@@ -0,0 +1,69 @@
+        validation: "text",
+        props: {
+          label: "D1.X.9 Plan overpayment reporting to the state",
+          hint: "Describe the plan’s latest annual overpayment recovery report submitted to the state as required under 42 CFR 438.608(d)(3).</br>Include, at minimum, the following information:<ul><li>The date of the report (rating period or calendar year).</li><li>The dollar amount of overpayments recovered.</li><li>The ratio of the dollar amount of overpayments recovered as a percent of premium ...
```
`plan's` instead of `plan’s`?

bangbay-bluetiger reviewed (approved):
Some suggestions, nothing blocking.

karla-vm commented:
@bangbay-bluetiger I addressed most of your comments in the latest commit, thanks for those suggestions 🥇 for the EntityType as a constant, I'd like to address that as a refactor ticket later on but I will make a note of it.
